example/glue/ObjectModel.hpp

```diff
@@ -80,32 +105,65 @@ class GC_ObjectModel
 	}
 
 	/**
+	 * Returns TRUE if an object is indexable, FALSE otherwise.
+	 *
+	 * @param objectPtr pointer to the object
+	 * @return TRUE if object is indexable, FALSE otherwise
+	 */
+	MMINLINE bool
+	isIndexable(omrobjectptr_t objectPtr)
+	{
+		return false;
+	}
+
+	/**
+	 * Set flags where ever they are
+	 * @param objectPtr Pointer to an object
+	 * @param bitsToClear mask to clear bits
+	 * @param bitsToSet mask to set bits
+	 */
+	MMINLINE void
+	setFlags(omrobjectptr_t objectPtr, uintptr_t bitsToClear, uintptr_t bitsToSet)
+	{
+		fomrobject_t* flagsPtr = (fomrobject_t*)SCAVENGER_FORWARDING_SLOT_EA(objectPtr);
+		fomrobject_t clear = (fomrobject_t)bitsToClear;
+		fomrobject_t set = (fomrobject_t)bitsToSet;
+
+		Assert_MM_true(clear <= OMR_OBJECT_HEADER_FLAGS_MASK);
+		Assert_MM_true(set <= OMR_OBJECT_HEADER_FLAGS_MASK);
+
+		*flagsPtr = (*flagsPtr & ~clear) | set;
+	}
+
+	/**
 	 * Returns TRUE if an object is dead, FALSE otherwise.
 	 * @param objectPtr Pointer to an object
 	 * @return TRUE if an object is dead, FALSE otherwise
 	 */
 	MMINLINE bool
 	isDeadObject(void *objectPtr)
 	{
-		return 0 != (*((uintptr_t *)objectPtr) & J9_GC_OBJ_HEAP_HOLE_MASK);
```

Comment:
I am not sure this should be using the scavenger forwarding slot as this code is used by all collectors